### DISCUSSION TITLE
Error running gisola with event and no waveform plot

### DISCUSSION THREAD

**Discussion by vdionicio** [nikosT/Gisola#30] [General]
Dear @nikosT,

I'm tring to run gisola with a particular event like this:

./gisola.py --event SGC2022dugzug -c ../test/config_sgc.yaml -l log_cron

the program is running well but at the end it gave me this error:

2022-02-23 19:49:48,261 INFO Function 'calculateInversions' executed 290.32 seconds
/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/site-packages/obspy/io/quakeml/core.py:1109: UserWarning: 'smi:org.gfz-potsdam.de/geofon/RMHP(10)>>BW(4,1,5)' is not a valid QuakeML URI. It will be in the final file but note that the file will not be a valid QuakeML file.
  warnings.warn(msg % obj.id)
2022-02-23 19:49:51,872 INFO Function 'gatherResults' executed 3.61 seconds
2022-02-23 19:49:58,946 ERROR Fatal Error occurred
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/home/sgc-geo46333/Gisola-main/src/plot.py", line 943, in caller
    func()
  File "/home/sgc-geo46333/Gisola-main/src/plot.py", line 430, in streams
    tr=config.st.select(station=sta, channel=cha)[0]
  File "/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/site-packages/obspy/core/stream.py", line 649, in __getitem__
    return self.traces.__getitem__(index)
IndexError: list index out of range
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/sgc-geo46333/Gisola-main/src/./gisola.py", line 236, in <module>
    plot.allplots()
  File "/home/sgc-geo46333/Gisola-main/src/config.py", line 181, in inner
    func(*args, **kwargs)
  File "/home/sgc-geo46333/Gisola-main/src/plot.py", line 958, in allplots
    res=p.map(caller, functions)
  File "/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/multiprocessing/pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/multiprocessing/pool.py", line 771, in get
    raise self._value
IndexError: list index out of range
2022-02-23 19:49:58,947 INFO Moving to next event, if any

When I open the list_all.html and get into the solution, I can not see the the waveforms image

![Captura de pantalla de 2022-02-23 15-12-59](https://user-images.githubusercontent.com/99366636/155400461-c9852c68-e98a-4bdb-a633-71849ff3c871.png)

I hope you can give suggestions to solve the problem.

Thanks a lot.

Best wishes,

Viviana

**Comment by nikosT** [comment]
Dear @vdionicio,

This looks quite strange! 
From the error message, it seems that there is a kind of inconsistency in the list with stations.
 
1. Is there enough space in the machine that runs Gisola? 
2. Does this happen with every event you tried? 
3. Does the test event (included in repo) work?
4. Could you share the output folder (send me in private message, if you think that it should not be public), including the log file?
5. Could you also send me the config.yaml file (don't forget to remove hostname/usernames/passwords in terms of privacy)?

**Comment by vdionicio** [comment]
Dear @nikosT,

Thanks a lot for your answer, to clarify the error, I wanna answer your questions:

1. Is there enough space in the machine that runs Gisola?
Yes, I'm runing gisola in a machine with 100G, and it has 80G available, 16 cores xeon, and 32 Gb ram.

2. Does this happen with every event you tried?

with my events yes, I run anotherone like: ./gisola.py --event SGC2019zgudxj -c ../test/config_sgc.yaml -l log_mesetas

and I got this error:
2022-02-25 21:52:21,487 INFO Function 'calculateInversions' executed 102.62 seconds
2022-02-25 21:52:29,637 INFO Function 'gatherResults' executed 8.15 seconds
2022-02-25 21:52:47,539 ERROR Fatal Error occurred
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/home/sgc-geo46333/Gisola-main/src/plot.py", line 943, in caller
    func()
  File "/home/sgc-geo46333/Gisola-main/src/plot.py", line 430, in streams
    tr=config.st.select(station=sta, channel=cha)[0]
  File "/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/site-packages/obspy/core/stream.py", line 649, in __getitem__
    return self.traces.__getitem__(index)
IndexError: list index out of range
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/sgc-geo46333/Gisola-main/src/./gisola.py", line 236, in <module>
    plot.allplots()
  File "/home/sgc-geo46333/Gisola-main/src/config.py", line 181, in inner
    func(*args, **kwargs)
  File "/home/sgc-geo46333/Gisola-main/src/plot.py", line 958, in allplots
    res=p.map(caller, functions)
  File "/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/multiprocessing/pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/multiprocessing/pool.py", line 771, in get
    raise self._value
IndexError: list index out of range
2022-02-25 21:52:47,540 INFO Moving to next event, if any


And I can't see the waveforms neither.


3.  Does the test event (included in repo) work?
Yes, I runit like: ./gisola.py -c ../test/config.yaml --event-xml ../test/benchmark/event.xml

and the results is:

2022-02-25 22:23:33,523 INFO Function 'calculateInversions' executed 427.14 seconds
/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/site-packages/obspy/io/quakeml/core.py:1109: UserWarning: 'smi:org.gfz-potsdam.de/geofon/RMHP(10)>>ITAPER(30)>>BW_BP(2,3.0,4)>>STALTA(1,30)' is not a valid QuakeML URI. It will be in the final file but note that the file will not be a valid QuakeML file.
  warnings.warn(msg % obj.id)
/home/sgc-geo46333/anaconda3/envs/gisola/lib/python3.9/site-packages/obspy/io/quakeml/core.py:1109: UserWarning: 'smi:org.gfz-potsdam.de/geofon/RMHP(10)>>ITAPER(10)>>BW(2,3.0,4)>>STALTA(1,30)' is not a valid QuakeML URI. It will be in the final file but note that the file will not be a valid QuakeML file.
  warnings.warn(msg % obj.id)
2022-02-25 22:23:37,612 INFO Function 'gatherResults' executed 4.09 seconds
2022-02-25 22:23:46,128 INFO Function 'allplots' executed 8.52 seconds


4.  Could you share the output folder (send me in private message, if you think that it should not be public), including the log file?
I have sent an email with this information, including the two events that I'm trying for this request in a .tar file

5. Could you also send me the config.yaml file (don't forget to remove hostname/usernames/passwords in terms of privacy)?
I have sent an email with this information, including the two events that I'm trying for this requestin a .tar file.  I'm attaching also the inventory file.

**Comment by nikosT** [comment]
Hi @vdionicio,

I see no e-mail. Could you send it here: ntriantafyl {@} cslab.ece.ntua.gr?

Also, just to be sure. Could you check that you have installed the suggested versions of the Python libraries as shown in environment.yml file?

**Comment by vdionicio** [comment]
Dear @Nikos,

I'm attaching the log and config,files, for the results folder and the
inventory file, I could not attach the tar file, but I'm sending a link
where you can download it, Let me know if you have any problems.

https://owncloudrsnc.sgc.gov.co/index.php/s/2t6pQB94UcMPIkF

Thanks a lot for your help,

With best regards,

Viviana

El mié, 23 feb 2022 a la(s) 19:34, Nikolaos Triantafyllis (
***@***.***) escribió:

> Dear @vdionicio <https://github.com/vdionicio>,
>
> This looks quite strange!
> From the error message, it seems that there is a kind of inconsistency in
> the list with stations.
>
>    1. Is there enough space in the machine that runs Gisola?
>    2. Does this happen with every event you tried?
>    3. Does the test event (included in repo) work?
>    4. Could you share the output folder (send me in private message, if
>    you think that it should not be public), including the log file?
>    5. Could you also send me the config.yaml file (don't forget to remove
>    hostname/usernames/passwords in terms of privacy)?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nikosT/Gisola/discussions/30#discussioncomment-2240351>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXWDN3ATFKZMNRWDYH2WTG3U4V4J3ANCNFSM5PFM6M6A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Viviana Dionicio

**Comment by nikosT** [comment]
@vdionicio, could you please share it in another way? I have already 5 failed attempts to download the zip file. 4.9 GB seems a pretty large event anyway. Could you share just one, with me?

**Comment by vdionicio** [comment]
Dear @Nikost,

Could you try to download the folder results_sgc_2 in the same link:

https://owncloudrsnc.sgc.gov.co/index.php/s/2t6pQB94UcMPIkF

There you can fins the results form the event of 2019 that I mentioned in
my previous comment in the foro.

Let me know if you succeeded,

With best regards,

Viviana

El vie, 25 feb 2022 a la(s) 19:18, Nikolaos Triantafyllis (
***@***.***) escribió:

> @vdionicio <https://github.com/vdionicio>, could you please share it in
> another way? I have already 5 failed attempts. 4.9 GB seems a pretty large
> event anyway. Could you share just one, with me?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nikosT/Gisola/discussions/30#discussioncomment-2254754>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXWDN3DG4VDVBAGUCDRDZ2TU5AL7FANCNFSM5PFM6M6A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Viviana Dionicio

**Comment by nikosT** [comment]
@vdionicio, failed again. Try google drive

**Comment by vdionicio** [comment]
ok, let's try with google drive:

https://drive.google.com/file/d/1iWHPxbhQZoA5wAsrvi1iTQVVegD-3xQE/view?usp=sharing

Thanks again,

Viviana

El sáb, 26 feb 2022 a la(s) 08:25, Nikolaos Triantafyllis (
***@***.***) escribió:

> @vdionicio <https://github.com/vdionicio>, failed again. Try google drive
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nikosT/Gisola/discussions/30#discussioncomment-2256419>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXWDN3FVCAE2QTJJBET3C3LU5DIF7ANCNFSM5PFM6M6A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Viviana Dionicio

**Comment by nikosT** [comment]
Hi @vdionicio,
I downloaded the dir successfully. I'm already looking at the error which seems very weird! Could you also send me the config.yaml file (without usernames/passwords)? Meanwhile, until I come back with a workaround, could you double-check that you are using the versions (especially obspy library) suggested at the [https://github.com/nikosT/Gisola/wiki/Installation](https://github.com/nikosT/Gisola/wiki/Installation)?

**Comment by vdionicio** [comment]
Good to hear the you could downloaded the file, I'm attaching the config
file. About the versions, I have installed:

- conda 4.11.0
- GNU Fortran (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
- I don't have nvfortran
- Gisola v1.0.1
- Docker version 20.10.12, build e91ed57

Thanks for trying,

Viviana

El sáb, 26 feb 2022 a la(s) 11:29, Nikolaos Triantafyllis (
***@***.***) escribió:

> Hi @vdionicio <https://github.com/vdionicio>,
> I downloaded the dir successfully. I'm already looking at the error which
> seems very weird! Could you also send me the config.yaml file (without
> usernames/passwords)? Meanwhile, until I come back with a workaround, could
> double-check that you are using the versions suggested at the
> https://github.com/nikosT/Gisola/wiki/Installation?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nikosT/Gisola/discussions/30#discussioncomment-2257030>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXWDN3DTUA6KNCK77ELNCFTU5D5XHANCNFSM5PFM6M6A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Viviana Dionicio

**Reply by nikosT** [reply]
> Good to hear the you could downloaded the file, I'm attaching the config file. About the versions, I have installed: - conda 4.11.0 - GNU Fortran (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0 - I don't have nvfortran - Gisola v1.0.1 - Docker version 20.10.12, build e91ed57 Thanks for trying, Viviana El sáb, 26 feb 2022 a la(s) 11:29, Nikolaos Triantafyllis ( ***@***.***) escribió:
> […](#)
> Hi @vdionicio <https://github.com/vdionicio>, I downloaded the dir successfully. I'm already looking at the error which seems very weird! Could you also send me the config.yaml file (without usernames/passwords)? Meanwhile, until I come back with a workaround, could double-check that you are using the versions suggested at the https://github.com/nikosT/Gisola/wiki/Installation? — Reply to this email directly, view it on GitHub <[#30 (comment)](https://github.com/nikosT/Gisola/discussions/30#discussioncomment-2257030)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AXWDN3DTUA6KNCK77ELNCFTU5D5XHANCNFSM5PFM6M6A> . Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. You are receiving this because you were mentioned.Message ID: ***@***.***>
> -- Viviana Dionicio

@vdionicio 
I see no attachments. Could you paste a link?

**Comment by vdionicio** [comment]
This is the link:
https://drive.google.com/file/d/1Cv-M3IFJmb53PqaYSTJI57-rU4o3ZPH7/view?usp=sharing



El sáb, 26 feb 2022 a la(s) 12:13, Nikolaos Triantafyllis (
***@***.***) escribió:

> Good to hear the you could downloaded the file, I'm attaching the config
> file. About the versions, I have installed: - conda 4.11.0 - GNU Fortran
> (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0 - I don't have nvfortran - Gisola
> v1.0.1 - Docker version 20.10.12, build e91ed57 Thanks for trying, Viviana
> El sáb, 26 feb 2022 a la(s) 11:29, Nikolaos Triantafyllis ( *@*.
>
> *) escribió: … <#m_-1524555309963475216_> Hi @vdionicio
> <https://github.com/vdionicio> https://github.com/vdionicio
> <https://github.com/vdionicio>, I downloaded the dir successfully. I'm
> already looking at the error which seems very weird! Could you also send me
> the config.yaml file (without usernames/passwords)? Meanwhile, until I come
> back with a workaround, could double-check that you are using the versions
> suggested at the https://github.com/nikosT/Gisola/wiki/Installation
> <https://github.com/nikosT/Gisola/wiki/Installation>? — Reply to this email
> directly, view it on GitHub <#30 (comment)
> <https://github.com/nikosT/Gisola/discussions/30#discussioncomment-2257030>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AXWDN3DTUA6KNCK77ELNCFTU5D5XHANCNFSM5PFM6M6A
> <https://github.com/notifications/unsubscribe-auth/AXWDN3DTUA6KNCK77ELNCFTU5D5XHANCNFSM5PFM6M6A>
> . Triage notifications on the go with GitHub Mobile for iOS
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
> You are receiving this because you were mentioned.Message ID: @.*>
> -- Viviana Dionicio
>
> @vdionicio <https://github.com/vdionicio>
> I see no attachments. Could you paste a link?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nikosT/Gisola/discussions/30#discussioncomment-2257170>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXWDN3APT3FFA52VG33CGGTU5EC4PANCNFSM5PFM6M6A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Viviana Dionicio

**Comment by nikosT** [comment]
@vdionicio, I also reproduce the problem. I'll let you know when it is fixed, as soon as possible.

**Comment by vdionicio** [comment]
Thanks a lot!!!

Viviana 

---
Viviana Dionicio 
Coordinadora Grupo Evaluación y Monitoreo de la Actividad Sísmica
Servicio Geológico Colombiano
Red Sismológica Nacional de Colombia
Ext: 2168
           
Diag. 53 N.° 34-53 Bogotá, Colombia                
(571) 2200 200 Ext. 2107                            

 
La información contenida en este mensaje y adjuntos es confidencial, para 
el uso exclusivo de la persona o entidad destinataria. Si usted ha recibido 
este mensaje por equivocación, se le notifica que no está autorizado para 
revisarlo, reenviarlo, imprimirlo, copiarlo o distribuirlo de forma parcial 
o total; por favor comunique el error a la persona que lo envía y elimínelo.

**Comment by nikosT** [comment]
Dear @vdionicio,

The error has been fixed. Could you download and try the latest Gisola release [v.1.1.0](https://github.com/nikosT/Gisola/releases/tag/v.1.1.0)?

Please let me know if this solved the issue.

By the way, if you run Gisola experimentally or operationally for your institute, we could add it to the use cases at the [readme](https://github.com/nikosT/Gisola#readme) (if you want to).

**Comment by vdionicio** [comment]
Thanks a lot for the help I’ll try with the new version and let you know. 

I work at the Colombian Geological Survey (Servicio Geológico Colombiano) and we manage the Colombian Seismological Network, I’m doing test to run the code in real time and if everything works well the idea is using it in daily-realtime operation. Of course you can include de the in the readme. 

Best Regards,

Viviana 

---

**Comment by vdionicio** [comment]
Dear @nikosT,

Now it's working for both events,

Thanks a lot,

Viviana

**Reply by nikosT** [reply]
@vdionicio, great!

We added the use case at [README](https://github.com/nikosT/Gisola/blob/main/README.md). In case you publish the page or you want to share a new screenshot with more events, please let us know. Thank you!
